### ISSUE TITLE
Feature Request: Support for R Notebook equivalent

### ISSUE THREAD

**Issue by ailich** [quarto-dev/quarto-cli#650]
Currently it seems that to render a Quarto document all of the code needs to be rerun. Are there plans to add something similar to [R Notebooks](https://argoshare.is.ed.ac.uk/healthyr_book/what-is-the-difference-between-a-notebook-and-an-r-markdown-file.html), which just use the last output from the chunks (In Rstudio this is accomplished with the "Preview" button when using an R notebook)? This can be nice for longer analyses or more interactive exercises where you may want to edit a plot or a later section but don't need to rerun all the earlier parts.

**Comment by jjallaire** [comment]
I think the best way to accomplish this for now is to use Knitr caching (https://quarto.org/docs/computations/caching.html#knitr-cache)

**Comment by ailich** [comment]
Thanks @jjallaire , I do still think there's a place for the "notebook" style though if there's time to add it in the future as it can be nice to not have to go around and flip all the chunk options (e.g. you can work interactively in the notebook to get things running and looking the way you want and then when you're done click knit/render to see if everything runs clean from start to finish). Also some packages don't always play nice with caching (e.g. raster and terra packages b/c they link to C++ objects/external files).

**Comment by jjallaire** [comment]
Yes, we will definitely be considering the pure notebook use case hopefully make some progress there soon.

**Comment by marianschmidt** [comment]
For me caching fails when trying to load large files from disk inside the .qmd file.

**Comment by GitHunter0** [comment]
> Yes, we will definitely be considering the pure notebook use case hopefully make some progress there soon.

`.qmd` should be the future R Notebook format, right?

**Comment by isaactpetersen** [comment]
Just wanted to state my strong support for a Notebook version of Quarto. This would allow working interactively on the Quarto script, which is important for data analysis because it often requires iterating. I almost exclusively use RMarkdown Notebooks (rather than Quarto) for this reason. Adding an interactive notebook feature to Quarto would address an important gap. I hope this will be supported.
